source/libNsdl/src/sn_grs.c

```diff
@@ -764,11 +763,8 @@ sn_nsdl_resource_info_s *sn_grs_search_resource(struct grs_s *handle, uint16_t p
     else if (search_method == SN_GRS_DELETE_METHOD) {
         /* Scan all nodes on list */
         ns_list_foreach(sn_nsdl_resource_info_s, resource_search_temp, &handle->resource_root_list) {
-            uint8_t *temp_ptr = resource_search_temp->path;
-
-            /* If found, return pointer */
-            if ((*(temp_ptr + (uint8_t)pathlen) == '/')
-                    && !memcmp(resource_search_temp->path, path_temp_ptr, pathlen)) {
+            if (resource_search_temp->pathlen > pathlen &&
+                    0 == memcmp(resource_search_temp->path, path, pathlen)) {
```

Comment:
Doesn't this return also resources with same prefix, eg. "dr/x" would find also "dr/xyz/1"?